src/utils/json-data.js

```diff
@@ -49,31 +48,53 @@ const buildGlobalDataFromJson = () => {
       throw new shared.StachePluginError(error.message);
     }
 
-    acc[propertyName] = JSON.parse(contents);
+    buildJsonDataObject(acc, propertyNameSegments, JSON.parse(contents));
 
     return acc;
   }, {});
 
   return dataObject;
 };
 
-const convertFileNameToObjectPropertyName = (fileName) => {
-  return fileName.split('.')[0]
-    .replace(/\s+/g, '_')     // Replace spaces with underscores
-    .replace(/[^\w-]+/g, '')  // Remove all non-word chars
-    .replace(/-/g, '_')       // Replace all dashes with underscores
-    .replace(/--+/g, '_')     // Replace multiple dashes with single underscore
-    .replace(/^-+/, '')       // Trim - from start of text
-    .replace(/-+$/, '');      // Trim - from end of text;
+const convertFileNameToObjectPropertyNameSegments = (fileName) => {
+  fileName =  fileName.split('.')[0]
+    .replace(/\\+/g, '.')      // Replace back-slashes with a single period
+    .replace(/\/+/g, '.')      // Replace slashes with a single period
+    .replace(/\s+/g, '_')      // Replace spaces with underscores
+    .replace(/[^.\w-]+/g, '')  // Remove all non-word chars other than periods
+    .replace(/-/g, '_')        // Replace all dashes with underscores
+    .replace(/--+/g, '_')      // Replace multiple dashes with single underscore
+    .replace(/^-+/, '')        // Trim - from start of text
+    .replace(/-+$/, '');       // Trim - from end of text;
+
+  // Split up nested property name in to an array of valid keys and return
+  return fileName.split('.').filter(key => { return !!key });
 };
 
 const isPropertyNameValid = (propertyName) => {
-  if (!propertyName || propertyName === 'prototype') {
+  if (!propertyName || !propertyName.length || propertyName.indexOf('prototype') > -1) {
     return false;
   }
 
   // Parsing as boolean because reserved-words returns `undefined` for falsy values.
-  return !reserved.check(propertyName, 'es6', true);
+  let valid = true;
+  propertyName.map(key => {
+    if (reserved.check(key, 'es6', true)) {
+      valid = false;
+    }
+  });
+  return valid;
+};
+
+const buildJsonDataObject = (baseObject, dataAttrNames, dataValue) => {
```

Comment:
I want to make sure I understand this correctly.  The purpose of this code is so that a file path like `stache/data/parent/child1/child2/custom.json` is accessible via `stache.jsonData.parent.child1.child2.custom`?

Reply:
Yep! The method here takes the array we have created and then builds a json data tree that can be accessed as you've stated.